File: cn/java-spring-demo/src/main/java/com/itangcent/springboot/demo/common/constant/UserType.java
package com.itangcent.springboot.demo.common.constant;

public enum UserType {
    ADMIN(1),
    MEMBER(2),
    GUEST(3);

    private final int type;

    UserType(int type) {
        this.type = type;
    }

    public int getType() {
        return type;
    }
} 

File: cn/kotlin-spring-demo/src/main/kotlin/com/itangcent/common/constant/UserType.kt
package com.itangcent.common.constant

/**
 * 用户类型枚举
 * 定义系统中所有可能的用户类型
 */
enum class UserType(val type: Int) {
    /**
     * 管理员用户
     * 具有系统最高权限
     */
    ADMIN(1),

    /**
     * 普通会员用户
     * 具有基本的系统使用权限
     */
    MEMBER(2),

    /**
     * 访客用户
     * 具有最低限度的系统访问权限
     */
    GUEST(3)
} 